### PR TITLE
Network selector: setup for custom headers 

### LINKED ISSUE #948: Improve Lab Network Selector
### What problem does your feature solve?
We want to support the ability to input API keys in the network selector

https://www.figma.com/design/yuikaFOPmJifyAfS90ER6T/Laboratory?node-id=1223-10254&t=BTmHcVTW3zAmR8rd-1

### PR DESCRIPTION
**Pull request by quietbits**
- Refactored Network Selector to allow custom headers
- Commented out custom header fields until https://github.com/stellar/js-stellar-sdk/issues/1088 is ready
  - Issue created: https://github.com/stellar/laboratory/issues/1133

### CODE REVIEW

**Comment by quietbits** [comment]
Waiting on a fix in JS Stellar SDK

File: src/app/(sidebar)/transaction/submit/page.tsx [hunk 1]
```diff
@@ -73,7 +74,9 @@ export default function SubmitTransaction() {
   const [isSaveTxnModalVisible, setIsSaveTxnModalVisible] = useState(false);
   const [isDropdownActive, setIsDropdownActive] = useState(false);
   const [isDropdownVisible, setIsDropdownVisible] = useState(false);
-  const [submitMethod, setSubmitMethod] = useState("");
+  const [submitMethod, setSubmitMethod] = useState<"horizon" | "rpc" | string>(
```
Comment (jeesunikim) [inline src/app/(sidebar)/transaction/submit/page.tsx:77]: aren't `"horizon" | "rpc"` suffice, @quietbits?

File: src/app/(sidebar)/transaction/submit/page.tsx [hunk 2]
```diff
@@ -178,21 +181,30 @@ export default function SubmitTransaction() {
...
-        networkPassphrase: network.passphrase,
-      });
-    } else {
-      // Do nothing
-    }
+    resetSubmitState();
+
+    delayedAction({
+      action: () => {
+        if (submitMethod === "rpc") {
```
Comment (jeesunikim) [inline src/app/(sidebar)/transaction/submit/page.tsx:188]: because i only see "rpc" and "horizon" here

File: src/query/useLatestLedger.ts [hunk 3]
```diff
@@ -1,12 +1,19 @@
+import { NetworkHeaders } from "@/types/types";
 import { SorobanRpc } from "@stellar/stellar-sdk";
```
Comment (jeesunikim) [inline src/query/useLatestLedger.ts:2]: should we use `rpc.Server` https://stellar.github.io/js-stellar-sdk/module-rpc.Server.html#.Options

Comment (quietbits) [reply src/query/useLatestLedger.ts:2]: Yes, I'll update. :+1:

File: src/helpers/getNetworkHeaders.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { EmptyObj, Network } from "@/types/types";
+
+export const getNetworkHeaders = (
+  network: Network | EmptyObj,
+  method: "horizon" | "rpc",
+) => {
+  if (method === "rpc" && network.rpcHeaderName) {
+    return { [network.rpcHeaderName]: network.rpcHeaderValue || "" };
+  } else if (method === "horizon" && network.horizonHeaderName) {
```
[the remark this answers is not in the document]
Comment (quietbits) [reply src/helpers/getNetworkHeaders.ts:9]: I just want to future-proof these cases. If we add another method in the future, the `else` case would also include that, but we probably won't want it.

**Review by jeesunikim — APPROVED** [review]

**Pull request merged**